infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationManager.java

```diff
@@ -189,20 +189,28 @@ public void processStandbyChange(TopologyDescriptor newConfig) {
         Set<String> standbysToRemove = new HashSet<>(currentStandbys);
         standbysToRemove.removeAll(intersection);
 
-        //Remove standbys that are not in the new config
+        // Remove standbys that are not in the new config
         for (String clusterId : standbysToRemove) {
             stopLogReplicationRuntime(clusterId);
             topology.removeStandbyCluster(clusterId);
         }
 
-        //Start the standbys that are in the new config but not in the current config
+        // Start the standbys that are in the new config but not in the current config
         for (String clusterId : newStandbys) {
             if (!runtimeToRemoteCluster.containsKey(clusterId)) {
                 ClusterDescriptor clusterInfo = newConfig.getStandbyClusters().get(clusterId);
                 topology.addStandbyCluster(clusterInfo);
                 startLogReplicationRuntime(clusterInfo);
             }
         }
+
+        // The connection id or other transportation plugin's info could've changed for
+        // existing standby cluster's, updating the routers will re-establish the connection
+        // to the correct endpoints/nodes
+        for (String clusterId : intersection) {
+            ClusterDescriptor clusterInfo = newConfig.getStandbyClusters().get(clusterId);
+            runtimeToRemoteCluster.get(clusterId).updateRouterClusterDescriptor(clusterInfo);
```

Comment:
Would this call be invoked over the network? I am trying to evaluate if we should maintain the cluster id s with changes and invoke this call only to those set of clusters vs sending the update to all the cluster ids.